src/schema/posts.ts

```diff
@@ -2314,6 +2349,107 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         },
       );
     },
+    startPostBoost: async (
+      _,
+      { postId, duration, budget }: Omit<StartPostBoostArgs, 'userId'>,
+      ctx: AuthContext,
+    ): Promise<GQLEmptyResponse> => {
+      if (duration < 1 || duration > 30 || duration % 1 !== 0) {
+        throw new ValidationError(
+          'Boost duration must be between 1 and 30 days',
+        );
+      }
+
+      if (budget < 1000 || budget > 100000 || budget % 1000 !== 0) {
+        throw new ValidationError(
+          'Boost budget must be between 1000 and 1000000',
+        );
+      }
+
+      const { userId, con } = ctx;
+      const [post, balance] = await Promise.all([
+        con.getRepository(Post).findOneOrFail({
+          select: ['id', 'flags'],
+          where: [
+            { id: postId, authorId: userId },
+            { id: postId, scoutId: userId },
+          ],
+        }),
+        getBalance({ userId }),
+      ]);
+
+      const total = budget * duration;
+
+      if (post.flags?.boosted) {
+        throw new ValidationError('Post is already boosted');
+      }
+
+      if (total > balance.amount) {
+        throw new ValidationError(
+          `Insufficient balance. You have ${balance.amount} available.`,
+        );
+      }
+
+      await ctx.con.transaction(async (entityManager) => {
+        const { campaignId } =
+          await skadiPersonalizedDigestClient.startPostCampaign({
```

Comment:
@capJavert I am thinking maybe to introduce a new instance of `SkadiClient` that's going to handle these new endpoints for post boost. Wdyt?

Reply:
If its the same service, you can just add additional method to `SkadiClient`. That said you can instantiate a new one for post boost to avoid using `skadiPersonalizedDigestClient`.

Reply:
Cool, let's do that.